modules/text2vec-aws/vectorizer/class_settings.go

```diff
@@ -192,6 +193,21 @@ func (ic *classSettings) TargetVariant() string {
 	return ic.getStringProperty(targetVariantProperty, "")
 }
 
+func (ic *classSettings) Dimensions() *int {
+	val, ok := ic.cfg.ClassByModuleName("text2vec-aws")[dimensionsProperty]
+	if !ok || val == nil {
+		return nil
+	}
+	if intVal, ok := val.(int); ok {
+		return &intVal
+	}
+	if floatVal, ok := val.(float64); ok {
+		intVal := int(floatVal)
+		return &intVal
+	}
+	return nil
```

Comment:
Let's use built-in methods here for getting the settings from modules

let's add `var defaultDimensions = int64(1536)` here and refactor this method to this:

```
func (ic *classSettings) Dimensions() *int64 {
	return ic.BaseClassSettings.GetPropertyAsInt64(dimensionsProperty, &defaultDimensions)
}
```

This method will always set dimensions to 1536 if it's not set.